src/uct/ib/rc/accel/rc_mlx5_devx.c

```diff
@@ -99,42 +91,83 @@ ucs_status_t uct_rc_mlx5_devx_iface_init_events(uct_rc_mlx5_iface_common_t *ifac
         ucs_error("mlx5dv_devx_create_event_channel() failed: %m");
         status = UCS_ERR_IO_ERROR;
         goto err;
-
     }
 
     status = ucs_sys_fcntl_modfl(event_channel->fd, O_NONBLOCK, 0);
     if (status != UCS_OK) {
         goto err_destroy_channel;
     }
 
-    status = ucs_async_set_event_handler(iface->super.super.super.worker->async->mode,
-                                         event_channel->fd, UCS_EVENT_SET_EVREAD,
-                                         uct_rc_mlx5_devx_iface_event_handler, iface,
-                                         iface->super.super.super.worker->async);
-    if (status != UCS_OK) {
-        goto err_destroy_channel;
+    if (create_async_handler) {
+        status = ucs_async_set_event_handler(
+                iface->super.super.super.worker->async->mode,
+                event_channel->fd, UCS_EVENT_SET_EVREAD,
+                uct_rc_mlx5_devx_iface_event_handler, iface,
+                iface->super.super.super.worker->async);
+        if (status != UCS_OK) {
+            goto err_destroy_channel;
+        }
     }
 
-    iface->event_channel = event_channel;
+    *event_channel_p = event_channel;
     return UCS_OK;
 
 err_destroy_channel:
     mlx5dv_devx_destroy_event_channel(event_channel);
-    iface->event_channel = NULL;
 err:
+    return status;
+}
+#endif
+
+ucs_status_t
+uct_rc_mlx5_devx_iface_init_events(uct_rc_mlx5_iface_common_t *iface)
+{
+    ucs_status_t status = UCS_OK;
+#if HAVE_DECL_MLX5DV_DEVX_SUBSCRIBE_DEVX_EVENT
+    uct_ib_mlx5_md_t *md;
+
+    iface->event_channel    = NULL;
+    iface->cq_event_channel = NULL;
+    md = ucs_derived_of(uct_ib_iface_md(&iface->super.super), uct_ib_mlx5_md_t);
```

Comment:
maybe add macro uct_ib_iface_mlx5_md, there are few relevant places that can use it